src/Config/SDK/ComponentProvider/Logs/LogRecordExporterConsole.php

```diff
@@ -31,6 +32,6 @@ public function createPlugin(array $properties, Context $context): LogRecordExpo
 
     public function getConfig(ComponentProviderRegistry $registry): ArrayNodeDefinition
     {
-        return new ArrayNodeDefinition('console');
+        return (new ArrayNodeDefinition('console'))->attribute(TreatNullAsUnsetNormalization::ALLOW_EMPTY, true);
```

Comment:
We should remove the `TreatNullAsUnsetNormalization` completely, the spec was changed in https://github.com/open-telemetry/opentelemetry-specification/pull/4269 from

> Parse MUST interpret null as equivalent to unset.

to

> Parse MUST differentiate between properties that are missing and properties that
are present but null.

Reply:
I originally did remove this normalization. It mostly worked, except for the case of not-set environment variables. Since they were no longer removed, the config was invalid. Perhaps we can say that's acceptable and it will fail-fast if your config refers to a not-set env var?

Reply:
We should not fail on not-set env vars as this would complicate migrating from env vars to `sdk-migration-config.yaml`.
See also https://github.com/open-telemetry/opentelemetry-configuration?tab=readme-ov-file#required-and-null-properties:

> - [type of null](https://json-schema.org/understanding-json-schema/reference/null): When a property includes a type of null along with other allowed types (i.e. "type": ["string", "null"]), it indicates that even if the property key is present, the value may be omitted. This is useful in a variety of situations:
>   - When modeling properties with primitive types which are candidates for [env var substitution](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution), since allowing null means that the configuration is valid even if the referenced env var is undefined.

A possible solution would be to override the `{...}Node` and `{...}NodeDefinition` classes to allow `null` values; early WIP in https://github.com/Nevay/otel-sdk-configuration/compare/v0.1.3...main.

Reply:
> We could add a component provider with a PackageDependency on open-telemetry/contrib-aws, similar to other propagators that are also published in different packages

I've opted to "mock" xray under `/tests`, reasoning that contrib components can supply their own ComponentProvider configuration...

Reply:
> A possible solution would be to override the {...}Node and {...}NodeDefinition classes to allow null values

I've copied across some of the code from Nevay/otel-sdk-configuration and it seems pretty close to working. One issue I cannot work out is that tests (and example code) which try to process instrumentation configuration are failing with:
```
Symfony\Component\Config\Definition\Exception\InvalidConfigurationException: Invalid configuration for path "open_telemetry.instrumentation.php.example_instrumentation": Component "OpenTelemetry\API\Instrumentation\AutoInstrumentation\InstrumentationConfiguration" uses unknown provider "span_name", available providers are "example_instrumentation"
```
It's looks like it's mangled the instrumentation componentList such that the key (component name) is there twice. Indeed, if I alter the config to something like this, it will process it correctly:
```yaml
instrumentation:
  general:
  php:
    example_instrumentation:
      example_instrumentation:
        span_name: ${EXAMPLE_INSTRUMENTATION_SPAN_NAME:-example span}
```

Reply:
For instrumentation config we'll need a new component registry method with the same config structure as `::component()` without the limitation to a single provider.

`::componentList()` cannot be used as it expects a different config structure:
```yaml
providers:
- providerA:
- providerB:
```

Reply:
...added `::componentMap()` for instrumentation config.